src/Data/List/Lazy/Types.purs

```diff
@@ -120,6 +125,22 @@ instance foldableList :: Foldable List where
 
   foldMap f = foldl (\b a -> b <> f a) mempty
 
+instance foldableWithIndexList :: FoldableWithIndex Int List where
+  foldrWithIndex f b xs =
```

Comment:
This seems like we could extract a default implementation in terms of `foldlWithIndex`. What do you think?

Reply:
I'll try.